### PR TITLE
Analytic Updates - 01262021

### PR DESCRIPTION
**Pull request by MHaggis**
New Analytic Stories:
- Trusted Developer Utilities Proxy Execution
- Trusted Developer Utilities Proxy Execution MSBuild

New Analytics:

Trusted Developer Utilities Proxy Execution MSBuild
- Suspicious MSBuild Rename (Passed)
- Suspicious MSBuild Spawn (Passed)
- Suspicious msbuild path (Passed)

Trusted Developer Utilities Proxy Execution
- Suspicious microsoft workflow compiler rename (Passed)
- Suspicious microsoft workflow compiler usage (Passed)

Suspicious MSHTA Activity
- Suspicious mshta child process (Passed)
- Detect MSHTA Url in Command Line (Passed)
- Detect Rundll32 Inline HTA Execution (Passed)
- Suspicious mshta spawn (Passed)
- Detect mshta inline hta execution (Passed)

Ryuk Ransomware 
- NLTest Domain Trust Discovery (Passed)
- WBAdmin Delete System Backups (Passed)

### CODE REVIEW

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :x:
- diskUsage: 270336
- runDuration: 0.7020000000000001s
- scanCount: 12

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect moved msbuild execution :x:
- diskUsage: 339968
- runDuration: 0.937s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :x:
- diskUsage: 339968
- runDuration: 0.921s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect moved msbuild execution :x:
- diskUsage: 335872
- runDuration: 0.873s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :x:
- diskUsage: 344064
- runDuration: 0.961s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :x:
- diskUsage: 270336
- runDuration: 0.706s
- scanCount: 13

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :x:
- diskUsage: 270336
- runDuration: 0.6960000000000001s
- scanCount: 10

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :x:
- diskUsage: 335872
- runDuration: 0.9550000000000001s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect wmiprvse spawn msbuild :white_check_mark:
- diskUsage: 364544
- runDuration: 2.987s
- scanCount: 3192

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect moved msbuild execution :white_check_mark:
- diskUsage: 368640
- runDuration: 2.922s
- scanCount: 3192

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :white_check_mark:
- diskUsage: 294912
- runDuration: 2.599s
- scanCount: 3208

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect microsoft.workflow.compiler.exe rename :white_check_mark:
- diskUsage: 299008
- runDuration: 8.677s
- scanCount: 14230

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect microsoft.workflow.compiler.exe usage :white_check_mark:
- diskUsage: 360448
- runDuration: 9.65s
- scanCount: 14217

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect rundll32 executing hta with inline protocols :white_check_mark:
- diskUsage: 360448
- runDuration: 6.6290000000000004s
- scanCount: 9131

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta url in command line :white_check_mark:
- diskUsage: 348160
- runDuration: 6.5360000000000005s
- scanCount: 9131

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawn :white_check_mark:
- diskUsage: 360448
- runDuration: 6.605s
- scanCount: 9131

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

WBAdmin Delete System Backups :white_check_mark:
- diskUsage: 348160
- runDuration: 1.222s
- scanCount: 29

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

NLTest Domain Trust Discovery :x:
- diskUsage: 126976
- runDuration: 0.004s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

NLTest Domain Trust Discovery :x:
- diskUsage: 126976
- runDuration: 0.003s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :x:
- diskUsage: 376832
- runDuration: 0.9390000000000001s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :x:
- diskUsage: 368640
- runDuration: 0.917s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

NLTest Domain Trust Discovery :white_check_mark:
- diskUsage: 430080
- runDuration: 1.1500000000000001s
- scanCount: 4

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :x:
- diskUsage: 376832
- runDuration: 0.937s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect mshta spawning a process :white_check_mark:
- diskUsage: 397312
- runDuration: 11.790000000000001s
- scanCount: 17702

File: detections/endpoint/suspicious_msbuild_path.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  your endpoints into the `Endpoint` datamodel in the `Processes` node.
+type: ESCU
+references:
+- https://lolbas-project.github.io/lolbas/Binaries/Msbuild/
+- https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1127.001/T1127.001.md
+author: Michael Haag, Splunk
+search: '| tstats `security_content_summariesonly` count values(Processes.process_name)
+  as process_name values(Processes.process) as process min(_time) as firstTime max(_time)
+  as lastTime from datamodel=Endpoint.Processes where Processes.process_name=msbuild.exe
+  AND (Processes.process_path!=c:\\windows\\microsoft.net\\framework*\\v4.0.30319\\*)
```
Comment (P4T12ICK) [inline detections/endpoint/suspicious_msbuild_path.yml]: Can we make that independent from the version?

File: detections/endpoint/suspicious_msbuild_path.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  by Processes.dest Processes.parent_process Processes.user | `drop_dm_object_name(Processes)`
+  | `security_content_ctime(firstTime)` | `security_content_ctime(lastTime)`| `suspicious_msbuild_path_filter`'
+known_false_positives: Some legitimate applications may use a moved copy of msbuild,
+  triggering a false positive. Baselining of MSBuild usage is recommended to better
+  understand it's path usage.
+tags:
+  analytics_story:
+  - Trusted Developer Utilities Proxy Execution MSBuild
+  mitre_attack_id:
+  - T1127.001, T1036.003
```
Comment (P4T12ICK) [inline detections/endpoint/suspicious_msbuild_path.yml]: A array in yml format is:
mitre_attack_id:
- T1127.001
- T1036.003

File: detections/endpoint/suspicious_msbuild_rename.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  parent_process_name, process_name, OriginalFileName, process_path, CommandLine |
+  rename Computer as dest | `security_content_ctime(firstTime)`| `security_content_ctime(lastTime)`|
+  `suspicious_msbuild_rename_filter`'
+known_false_positives: Although unlikely, some legitimate applications may use a moved
+  copy of msbuild, triggering a false positive.
+tags:
+  analytics_story:
+  - Trusted Developer Utilities Proxy Execution MSBuild
+  mitre_attack_id:
+  - T1127.001, T1036.003
```
Comment (P4T12ICK) [inline detections/endpoint/suspicious_msbuild_rename.yml]: same as above

File: detections/endpoint/suspicious_msbuild_spawn.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  your endpoints into the `Endpoint` datamodel in the `Processes` node.
+type: ESCU
+references:
+- https://lolbas-project.github.io/lolbas/Binaries/Msbuild/
+- https://github.com/redcanaryco/atomic-red-team/blob/master/atomics/T1127.001/T1127.001.md
+author: Michael Haag, Splunk
+search: '| tstats `security_content_summariesonly` count values(Processes.process_name)
+  as process_name values(Processes.process) as process min(_time) as firstTime max(_time)
+  as lastTime from datamodel=Endpoint.Processes where Processes.parent_process_name=wmiprvse.exe
+  AND Processes.process_name=msbuild.exe by Processes.dest Processes.parent_process
```
Comment (P4T12ICK) [inline detections/endpoint/suspicious_msbuild_spawn.yml:23]: Only FYI: AND are not really needed in Splunk. A space is the same as an AND

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Detect renamed msbuild execution :white_check_mark:
- diskUsage: 299008
- runDuration: 2.657s
- scanCount: 3205

**Review by P4T12ICK — DISMISSED** [review]
Looks good to me

File: detections/endpoint/suspicious_msbuild_path.yml (hunk 1 above)
Comment (MHaggis) [reply detections/endpoint/suspicious_msbuild_path.yml]: Removed the specified version and updated false positive and description to reflect Visual Studio usage of `msbuild.exe`.

**Review by patel-bhavin — APPROVED** [review]

**Pull request merged**